### PR TITLE
Unify watchdog logic into a new IpmiWatchdog module.

### PR DESCRIPTION
**Pull request by cfernald**
- Create the new IpmiWatchdog module for handle FRB2 & OS watchdogs
- Adds PCD configuration options for setting IPMI watchdogs.
- Adds a mock implementation for the WDT IPMI calls.

### CODE REVIEW

**Review by apop5 — APPROVED** [review]

File: IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+    DEBUG_INFO,
+    "Setting IPMI watchdog timer. Timer: 0x%x Action: 0x%x ClearFlags: 0x%x Countdown: 0x%x\n",
+    TimerUse,
+    TimerAction,
+    FlagsClear,
+    CountdownValue
+    ));
+
+  ASSERT (TimerUse != 0);
+  ASSERT (CountdownValue != 0);
```
Comment (os-d) [inline IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c:57]: In the production FW case, should this function error out in these cases or is it safe for it to continue?

File: IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  }
+
+  //
+  // Initialize the settings.
+  //
+  ZeroMem (&WatchdogTimer, sizeof (WatchdogTimer));
+  WatchdogTimer.TimerUse.Bits.TimerUse          = TimerUse;
+  WatchdogTimer.TimerActions.Bits.TimeoutAction = TimerAction;
+  WatchdogTimer.TimerUseExpirationFlagsClear    = FlagsClear;
+  WatchdogTimer.InitialCountdownValue           = CountdownValue * 10; // Convert from 100ms to seconds.
```
Comment (os-d) [inline IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c:72]: This is actually converting seconds to 100ms, right?

**Review by os-d — APPROVED** [review]

File: IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c (hunk 1 above)
Comment (cfernald) [reply IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c:57]: Timer use should be statcally defined so this is just sanity, but i'll codify this to be safe.

File: IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c (hunk 2 above)
Comment (cfernald) [reply IpmiFeaturePkg/IpmiWatchdog/IpmiWatchdog.c:72]: Yes, thanks!

**Pull request merged**